daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -6,40 +6,28 @@ package crypto
 
 import java.nio.ByteBuffer
 import java.security.{MessageDigest, SecureRandom}
-import java.util
 import java.util.concurrent.atomic.AtomicLong
 
-import com.digitalasset.daml.lf.data.{ImmArray, Ref, Time, Utf8}
+import com.digitalasset.daml.lf.data.{Bytes, ImmArray, Ref, Time, Utf8}
 import com.digitalasset.daml.lf.value.Value
 import javax.crypto.Mac
 import javax.crypto.spec.SecretKeySpec
 
 import scala.util.control.NoStackTrace
 
-final class Hash private (private val bytes: Array[Byte]) {
+final class Hash private (val bytes: Bytes) {
```

Comment:
And this; can it be `AnyVal`? You'd get rid of `equals` and `hashCode` implementations.

Reply:
same as before. It cannot because Bytes is already a user-defined `AnyVal`

Reply:
Same as for `Offset`: `Bytes` already is an `AnyVal`.